src/Tokenizer/Tokens.php

```diff
@@ -1050,7 +1050,12 @@ public function setCode($code)
         // clear memory
         $this->setSize(0);
 
-        $tokens = \defined('TOKEN_PARSE')
+        /**
+         * T_NULLSAFE_OBJECT_OPERATOR will not appear if TOKEN_PARSE is used on PHP 8.0.0.
+         *
+         * @see https://bugs.php.net/bug.php?id=80462
+         */
+        $tokens = \defined('TOKEN_PARSE') && \PHP_VERSION_ID !== 80000
```

Comment:
frankly speaking, i would rather disallow to run the tool on v8.0.0 than pollute source with this hack to workaround language bug.
keep in mind that we DEPENDS on usage of TOKEN_PARSE - if the language is high enough to have it, it means it actually have a syntax changes that TOKEN_PARSE is supporting. dropping TOKEN_PARSE will change behaviour not only for T_NULLSAFE_OBJECT_OPERATOR, but also for other tokens

Reply:
Indeed now I understand why `Preg::match(` is rewritten as `Preg::match (` :confounded: 

Can we really disallow PHP-CS-Fixer to run on PHP 8.0.0?

Reply:
Okay I guess it has to do with https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/563945b979a9850a4c6c5024b77c27d6e02b0eb1/php-cs-fixer#L26

As long as PHP-CS-Fixer does not officially support PHP 8 I guess there is nothing to do?

Then

```diff
- } elseif (!defined('PHP_VERSION_ID') || \PHP_VERSION_ID < 50600 || \PHP_VERSION_ID >= 70500) {
+ } elseif (!defined('PHP_VERSION_ID') || \PHP_VERSION_ID < 50600 || \PHP_VERSION_ID >= 80100 || PHP_VERSION_ID === 80000) {
```
?

Reply:
you would need to add a check in bootstrap file (`php-cs-fixer` in repo root folder)